examples/basic_operations/get_responsive_search_ads.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This example gets non-removed responsive search ads in a specified ad group.
+
+To add responsive search ads, run basic_operations/add_responsive_search_ad.py.
+To get ad groups, run basic_operations/get_ad_groups.py.
+"""
+
+import argparse
+import sys
+
+from google.ads.google_ads.client import GoogleAdsClient
+from google.ads.google_ads.errors import GoogleAdsException
+
+
+_DEFAULT_PAGE_SIZE = 1000
+
+
+def main(client, customer_id, page_size, ad_group_id=None):
+    ga_service = client.get_service('GoogleAdsService', version='v2')
+
+    query = ('SELECT ad_group.id, ad_group_ad.ad.id, '
+             'ad_group_ad.ad.responsive_search_ad.headlines, '
+             'ad_group_ad.ad.responsive_search_ad.descriptions, '
+             'ad_group_ad.status FROM ad_group_ad '
+             'WHERE ad_group_ad.ad.type = RESPONSIVE_SEARCH_AD '
+             'AND ad_group_ad.status != ''REMOVED''')
+
+    if ad_group_id:
+        query = query + f' AND ad_group.id = {ad_group_id}'
+
+    results = ga_service.search(customer_id, query=query, page_size=page_size)
+    aga_status_enum = client.get_type(
+        'AdGroupAdStatusEnum', version='v2').AdGroupAdStatus
+
+    try:
+        one_found = False
+
+        for row in results:
+            one_found = True
+            ad = row.ad_group_ad.ad
+            print(f'Responsive search ad with resource name '
+                  f'"{ad.resource_name}", '
+                  f'status {aga_status_enum.Name(row.ad_group_ad.status)} '
+                  f'was found.')
+            if ad.responsive_search_ad:
+                responsive_search_ad_info = ad.responsive_search_ad
+                print('Headlines:\n{}\n'
+                      'Descriptions:\n{}\n'.format(
+                          '\n'.join(_ad_text_assets_to_strs(
+                              client, responsive_search_ad_info.headlines)),
+                          '\n'.join(_ad_text_assets_to_strs(
+                              client, responsive_search_ad_info.descriptions))))
+            else:
+                print('\tResponsive search ad info was not found.')
+
+        if not one_found:
```

Comment:
No, `results` is a [GRPC iterator](https://googleapis.dev/python/google-api-core/latest/page_iterator.html) which is not a falsey. Data is returned lazily so the only way to know if there are >0 results is to start iterating. I'd rather leave in `one_found` because I think it is important to illustrate and show a message when there are zero results rather than ending silently.

Reply:
Got it. Thanks.